### PR TITLE
Adds support for HB checkins.

### LINKED ISSUE #2055: use Honeybadger checkins to ensure crons are running
See https://docs.honeybadger.io/guides/check-ins/

If feasible, this approach would be a fine way to ensure our cron jobs are running:

See 
- config/schedule.rb, 
- production crontab (`crontab -l` on correct VM for crons)
- dashboard audit information

I think:
- MoabToCatalog (m2c)
- CatalogToMoab (c2m)
- ChecksumValidation
- CatalogToArchive (which has 4 redis queues, fwiw)

### PR DESCRIPTION
**Pull request by justinlittman**
closes #2055

## Why was this change made? 🤔
To make sure that critical cron jobs run.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, or if it changes code used internally for cloud replication, ***run [integration test preassembly_image_accessioning_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/preassembly_image_accessioning_spec.rb) against stage as it tests preservation***, and/or test in stage environment, in addition to specs. The main classes relevant to replication are `ZipmakerJob`, `PlexerJob`, `*DeliveryJob`, `ResultsRecorderJob`, and `DruidVersionZip`; [see here for overview diagram of replication pipeline](https://github.com/sul-dlss/preservation_catalog/blob/main/app/jobs/README.md).⚡

QA

### CODE REVIEW

**Review by edsu — APPROVED** [review]
This is really nice to have!

File: README.md [hunk 1]
```diff
@@ -395,3 +395,12 @@ The token generated by `rake generate_token` should be passed along in the `Auth
...
 At present, all tokens grant the same (full) access to the read/update API.
+
+## Cron check-ins
+Some cron jobs (configured via the whenever gem) are integrated with Honeybadger check-ins. These cron jobs will check-in with HB (via a curl request to an HB endpoint) whenever run. If a cron job does not check-in as expected, HB will alert.
+
+Cron check-ins are configured in the following locations:
+1. `config/schedule.rb`: This specifies which cron jobs check-in and what setting keys to use for the checkin key. See this file for more details.
+2. `config/settings.yml`: Stubs out a check-in key for each cron job. Since we may not want to have a check-in for all environments, this stub key will be used and produce a null check-in.
+3. `config/settings/production.yml` in shared_configs: This contains the actual check-in keys.
+4. HB notification page: Check-ins are configured per project in HB. To configure a check-in, the cron schedule will be needed, which can be found with `bundle exec whenever`. After a check-in is created, the check-in key will be available. (If the URL is `https://api.honeybadger.io/v1/check_in/rkIdpB` then the check-in key will be `rkIdp`).
```
Comment (mjgiarlo) [inline README.md:406]: @justinlittman The end of the URL is `rkIdpB` and the check-in key is `rkIdp` (without the trailing `B`)... is that intended?

File: config/schedule.rb [hunk 2]
```diff
@@ -3,19 +3,31 @@
...
+  set :check_in, Settings.honeybadger_checkins.moab_to_catalog
+  runner_hb 'MoabStorageRoot.find_each(&:m2c_check!)'
 end
 
 # 11 am on the 15th of every month - the 'whenever' syntax for this is awkward and needs an ignored month
 # for the day to get parsed, so just use raw cron syntax
 every '0 11 15 * *', roles: [:queue_populator] do
   set :output, standard: nil, error: 'log/c2m-err.log'
-  runner 'MoabStorageRoot.find_each(&:c2m_check!)'
+  set :check_in, Settings.honeybadger_checkins.catalog_to_moab
```
Comment (ndushay) [inline config/schedule.rb:29]: add warning to ALL of these about changing HB checkin?  Or move warning to be seaparate from first job?

File: config/schedule.rb [hunk 3]
```diff
@@ -3,19 +3,31 @@
...
 
+require 'config'
+
+Config.load_and_set_settings(Config.setting_files('config', 'production'))
+
 # these cron jobs all append to existing log files
 
+# These define jobs that checkin with Honeybadger.
+# If changing the schedule of one of these jobs, also update at https://app.honeybadger.io/projects/54415/check_ins
+job_type :rake, "cd :path && :environment_variable=:environment bundle exec rake :task --silent :output && curl 'https://api.honeybadger.io/v1/check_in/:check_in"
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply config/schedule.rb:14]: It is just there to be an exemplar for other projects that are cribbing from pres cat.

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply config/schedule.rb:14]: @justinlittman Nice, thanks!

File: README.md (hunk 1 above)
Comment (edsu) [reply README.md:406]: I submitted a PR to fix this since I didn't catch it during review.

https://github.com/sul-dlss/preservation_catalog/pull/2100

**Pull request merged**